File: 217-contains-duplicate/contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> ans = new HashSet<Integer>();

        for(int n : nums){
            if(ans.contains(n)){
                return true;
            }
            ans.add(n);
        }
        return false;
    }
}

File: 217-contains-duplicate/contains-duplicate.cs
public class Solution {
    public bool ContainsDuplicate(int[] nums) {
        // HashSet<int> ans = new HashSet<int>();

        // foreach (int n in nums)
        //     if(!ans.Add(n))return true;
        
        // return false;
          HashSet<int> seen = new HashSet<int>();

        foreach (int num in nums) {
            if (seen.Contains(num)) {
                return true;
            }

            seen.Add(num);
        }

        return false;
    }
}